### PR TITLE
upload Logger.h

### PR DESCRIPTION
**Pull request by LinRenHong**

### CODE REVIEW

File: src/Header/Logger.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// Write operation and result log
+class Logger
+{
+    
+public:
+    
+    // Singleton code convention method
+    static unique_ptr<Logger> GetInstance();
+    
+    // Write data and warming message
```
Comment (john0312) [inline src/Header/Logger.h]: warning

File: src/Header/Logger.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    void WriteLog(const OperatorData &data, const std::string &level) const;
+    
+    // Not copyable or movable
+    Logger(const Logger &copy) = delete;
+    Logger& operator = (const Logger &copy) = delete;
+
+private:
+
```
Comment (john0312) [inline src/Header/Logger.h]: Extra new line, please remove.

File: src/Header/Logger.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+
```
Comment (john0312) [inline src/Header/Logger.h]: Extra new line, please remove.

File: src/Header/Logger.h [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+#define GROUP4_SRC_LOGGER_H_
+
+#include <string>
+#include <memory>
+#include "OperatorData.h"
+
+// Write operation and result log
+class Logger
+{
+
```
Comment (john0312) [inline src/Header/Logger.h]: Extra new line, please remove.

File: src/Header/Logger.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+#include <string>
+#include <memory>
+#include "OperatorData.h"
+
+// Write operation and result log
+class Logger
+{
+    
+public:
+
```
Comment (john0312) [inline src/Header/Logger.h]: Extra new line, please remove.

File: src/Header/Logger.h [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+class Logger
+{
+    
+public:
+    
+    // Singleton code convention method
+    static unique_ptr<Logger> GetInstance();
+    
+    // Write data and warming message
+    void WriteLog(const OperatorData &data, const std::string &level) const;
```
Comment (john0312) [inline src/Header/Logger.h]: Consider using enum for log level.

File: src/Header/Logger.h [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+#include "OperatorData.h"
+
+// Write operation and result log
+class Logger
+{
+    
+public:
+    
+    // Singleton code convention method
+    static unique_ptr<Logger> GetInstance();
```
Comment (john0312) [inline src/Header/Logger.h]: Please use raw pointer for singleton get method. There can only be one copy of unique_ptr so it won't work for singleton.

File: src/Header/Logger.h [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+// Write operation and result log
+class Logger
+{
+public:
+    // Singleton code convention method
+    static Logger* GetInstance();
+    
+    // Write data and warning message
+    void WriteLog(const OperatorData &data, const Level &level) const;
```
Comment (john0312) [inline src/Header/Logger.h:20]: Where is enum Level defined? Usually we define it in the same file or in the same class.

**Comment by LinRenHong** [comment]
Remove Logger module

**Pull request closed without merging**